api/src/Repository/UserRepository.php

```diff
@@ -34,9 +34,30 @@ public function upgradePassword(UserInterface $user, string $newEncodedPassword)
         $this->_em->flush();
     }
 
-    public function filterByUser(QueryBuilder $queryBuilder, User $user): void {
-        $rootAlias = $queryBuilder->getRootAliases()[0];
-        $queryBuilder->andWhere("{$rootAlias}.id = :current_user");
-        $queryBuilder->setParameter('current_user', $user->getId());
+    /**
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @throws \Doctrine\ORM\NoResultException
+     */
+    public function areUsersRelatedViaCamp(User $user1, User $user2): bool {
+        if ($user1 === $user2) {
+            return true;
+        }
+        $queryBuilder = $this->_em->createQueryBuilder();
+        $queryBuilder->select('count(user1)');
+        $queryBuilder->from(User::class, 'user1');
+        $queryBuilder->join('user1.collaborations', 'userCampCollaborations');
+        $queryBuilder->join('userCampCollaborations.camp', 'camp');
+        $queryBuilder->join('camp.collaborations', 'relatedCampCollaborations');
+        $queryBuilder->andWhere('user1 = :user1');
+        $queryBuilder->andWhere('userCampCollaborations.status = :status_established');
+        $queryBuilder->andWhere('relatedCampCollaborations.status = :status_established');
+        $queryBuilder->andWhere('relatedCampCollaborations.user = :user2');
+        $queryBuilder->setParameter('user1', $user1);
+        $queryBuilder->setParameter('user2', $user2);
+        $queryBuilder->setParameter('status_established', CampCollaboration::STATUS_ESTABLISHED);
```

Comment:
Beim `CampRoleVoter` überprüfe ich zusätzlich noch, ob die Rolle eine von `CampCollaboration::VALID_ROLES` ist. Ist zwar momentan nicht möglich dass etwas anderes in der DB gespeichert ist. Denkst du das wäre sinnvoll oder unnötig?

Reply:
Momentan ist es für diesen Check nicht wirklich wichtig, in welcher Rolle sie releated sind.
Hier ist vor allem der status wichtig, da wir die zugangsbeschränkung nicht auf rollen aufbauen.
Dass in der DB eine ungültige Rolle ist (oder dass ein Hacker es nicht schafft, sich eine gültige zu geben) halte ich für keinen user case.